### PR TITLE
Migration script from loading the file to the db tables

### PR DESCRIPTION
**Pull request by JingJZ160**
Script for loading the file to the db tables. Please run the script under the project root instead of the bin folder like `python bin/migrate_json_to_db.py` but make sure you edit the script to specify the task ids. Thought this could be more flexible than fully automate loading the whole `__task` folder.

There is something missing in the python library path but it seems that we need to do some bigger renaming and folder moving changes to fix it. Since this is a one-time thing (I hope), I'm just hard-coding the path. 

----Update----
Run with `python -m scripts.migrate_json_to_db` under the project root folder.

### CODE REVIEW

File: ar/data.py [hunk 1]
```diff
@@ -191,11 +191,12 @@ def fetch_annotation(task_id, user_id, ar_id):
         return None
 
 
-def fetch_all_annotation_ids(task_id, user_id):
+def fetch_all_ar_ids(task_id, user_id):
     '''
     Return a list of ar_id for this task that has been annotated by this user.
     '''
     _dir = os.path.join(_task_dir(task_id), DIR_ANNO, user_id)
+    print(_dir)
```
Comment (edu-gp) [inline ar/data.py:199]: remove

File: shared/config.py [hunk 2]
```diff
@@ -8,30 +8,35 @@ class Config:
 
     This module is designed to fail loudly when a required env var is not found.
     """
-
-    def get_backend_password():
+    @classmethod
```
Comment (edu-gp) [inline shared/config.py:11]: `@staticmethod` ? Since `cls` is not used.

File: bin/migrate_json_to_db.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import copy
```
Comment (edu-gp) [inline bin/migrate_json_to_db.py:1]: We already have a `scripts` folder, can we move this file under there?

File: bin/migrate_json_to_db.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+sys.path.append(".")
+
+from ar import fetch_all_ar_ids
+from ar.data import fetch_annotation, _get_all_annotators_from_annotated
+from db.config import DevelopmentConfig
+from db.model import Database, User, EntityType, Context, Entity, Label, \
+    ClassificationAnnotation
+
+ENTITY_TYPE_NAME = "company"
```
Comment (edu-gp) [inline bin/migrate_json_to_db.py:15]: Can we keep the entity type in an enum like https://github.com/edu-gp/annotation_tool/blob/master/db/model.py#L146

File: bin/migrate_json_to_db.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+def _convert_single_annotation(anno, username):
+    user = get_or_create(User, username=username)
+
+    entity_type = get_or_create(EntityType, name=ENTITY_TYPE_NAME)
+
+    annotation_context = json.dumps(anno["req"]["data"], sort_keys=True)
+    context = get_or_create(Context, exclude_keys_in_read=["data"],
+                            hash=hashlib.md5(
+                                annotation_context.encode()).hexdigest(),
```
Comment (edu-gp) [inline bin/migrate_json_to_db.py:35]: Let's refactor this hash function for reuse

File: ar/data.py (hunk 1 above)
Comment (JingJZ160) [reply ar/data.py:199]: Done.

File: shared/config.py (hunk 2 above)
Comment (JingJZ160) [reply shared/config.py:11]: Done.

File: bin/migrate_json_to_db.py (hunk 3 above)
Comment (JingJZ160) [reply bin/migrate_json_to_db.py:1]: Done.

File: bin/migrate_json_to_db.py (hunk 4 above)
Comment (JingJZ160) [reply bin/migrate_json_to_db.py:15]: Sure. We've already have `EntityType` so I'll name it as `EntityTypeEnum`.

File: bin/migrate_json_to_db.py (hunk 5 above)
Comment (JingJZ160) [reply bin/migrate_json_to_db.py]: Done.

**Pull request merged**